modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc

```diff
@@ -0,0 +1,226 @@
+kind: ClusterIssuer
+metadata:
+  name: knative-eventing-selfsigned-issuer
+spec:
+  selfSigned: {}
+----
++
+. Apply the `ClusterIssuer` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a root certificate using the previously created `SelfSigned` `ClusterIssuer`:
++
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: knative-eventing-selfsigned-ca
+  namespace: cert-manager <1>
+spec:
+  secretName: knative-eventing-ca <2>
+
+  isCA: true
+  commonName: selfsigned-ca
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+
+  issuerRef:
+    name: knative-eventing-selfsigned-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+
+----
++
+<1> The {certmanageroperatorname} namespace, `cert-manager` by default.
+<2> Secret name that will be later used for the `ClusterIssuer` for Eventing
++
+. Apply the `Certificate` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Setup `ClusterIssuer` for Eventing
+
+. Create the `knative-eventing-ca-issuer` `ClusterIssuer` for Eventing:
++
+[source,yaml]
+----
+# This is the issuer that every Eventing component should use to issue their server's certs.
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: knative-eventing-ca-issuer
+spec:
+  ca:
+    secretName: knative-eventing-ca <1>
+----
++
+<1> Secret name in the {certmanageroperatorname} namespace (`cert-manager` by default) containing the certificate that can then be used by Knative Eventing components for new certificates.
++
+[IMPORTANT]
+====
+The name of the `ClusterIssuer` must be `knative-eventing-ca-issuer`.
+====
++
+. Apply the `ClusterIssuer` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Understanding the transport encryption configuration
+
+The transport-encryption feature flag is an enum configuration that configures how Addressables (Broker, Channel, Sink) should or must accept events.
+
+The possible values for `transport-encryption` are:
+
+* *disabled* (this is equivalent to the current behavior)
+** Addressables may accept events to HTTPS endpoints
+** Producers may send events to HTTPS endpoints
+* *permissive*
+** Addressables should accept events on both HTTP and HTTPS endpoints
+** Addressables should advertise both HTTP and HTTPS endpoints
+** Producers should prefer sending events to HTTPS endpoints, if available
+* *strict*
+** Addressables must not accept events to non-HTTPS endpoints
+** Addressables must only advertise HTTPS endpoints
+
+== Setup transport encryption in `KnativeEventing`
+
+. Enabling transport-encryption in `KnativeEventing`:
++
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+
+  # Other spec fields omitted ...
+  # ...
+
+  config:
+    features:
+      transport-encryption: strict <1>
+----
++
+<1> Configure `transport-encryption` to `strict`.
```

Comment:
does this only work with `strict`?

Reply:
no, permissive process is the same, only the config is different